Feature/bundle injector

SoldoV opened the pull request:
Create bundleInjector utility.
There are two parameters in loadBundles method:
**1. Config**
 Example:
```
const config = {
    scriptType: 'meta',
    baseURL: process.env.VUE_APP_ASSETS_BASEURL,
    styles: false,
    scripts: true
};
```
1. scriptType could be ```components``` or ```meta```. Builder imports only meta scripts, while other two projects import only components scripts.
2. baseURL, only vue-cli-plugin doesn't have it so it should be an empty string ```baseURL: ''```. For other two it should be ```baseURL: process.env.VUE_APP_ASSETS_BASEURL```
3. styles - boolean value, inject all or no style sheets.
4. scripts - boolean value, inject all or no scripts

**2. Bundle** 
Object that contains all bundle information.

Example of the config object passed as a parameter in builder actions.js :
```
const config = {
    scriptType: 'meta',
    baseURL: process.env.VUE_APP_ASSETS_BASEURL,
    style: false,
    script: true
};
const bundlePromise = bundleInjector.loadBundles(config, bundle).then(() => {...
```
Ref https://github.com/nsftx/chameleon-sdk/issues/20

Lanchi commented:
Please write some kind of description, what is done regarding `loadBundles` method, what's expected to be in `config`...

SoldoV commented:
> Please write some kind of description, what is done regarding `loadBundles` method, what's expected to be in `config`...

I deleted the comment before creating the PR.
I imagined the config to be :
```
const config = {
    bundle,
    scriptType: 'meta',
    baseURL: process.env.VUE_APP_ASSETS_BASEURL,
    style: false,
    script: true
};
```

1. bundle is the ```bundle``` object and it should contain all bundle information.
2. scriptType could be ```components``` or ```meta```. Builder imports only meta scripts, while other two projects import only components scripts.
3. baseURL, only vue-cli-plugin doesn't have it so I guess it should be an empty string ```baseURL: ''```. For other two it should be ```baseURL: process.env.VUE_APP_ASSETS_BASEURL```
4. style - boolean value, inject all or no style sheets.
5. script - boolean value, inject all or no scripts

Example of the config object passed as a parameter in builder actions.js :
```
const config = {
    bundle,
    scriptType: 'meta',
    baseURL: process.env.VUE_APP_ASSETS_BASEURL,
    style: false,
    script: true
};
const bundlePromise = bundleInjector.loadBundles(config).then(() => {...```

Lanchi commented:
>  >Please write some kind of description, what is done regarding loadBundles method, what's expected to be in config...
>
> I deleted the comment in the code before creating the PR.

I meant the PR description. PR should always (unless it's a simple obvious single commit) have a description so reviewers get the context and for later reference, if needed.
Other than that, you should document `loadBundle` method in JSDoc manner.

Let's keep `bundle` and `config` as a separate parameters - `bundle` is a _what_ and `config` is a _how_. It would be nicer if `style` and `script` inside `config` is in plural so it's consistent with bundle definition.

One of the requested items was
> inject all or no scripts

how is this done current implementation?

SoldoV commented:
> > > Please write some kind of description, what is done regarding loadBundles method, what's expected to be in config...
> > 
> > 
> > I deleted the comment in the code before creating the PR.
> 
> I meant the PR description. PR should always (unless it's a simple obvious single commit) have a description so reviewers get the context and, if needed, for later reference.
> Other than that, you should document `loadBundle` method in JSDoc manner.

I have edited the PR description and documented loadBundles method.

> 
> Let's keep `bundle` and `config` as a separate parameters - `bundle` is a _what_ and `config` is a _how_. It would be nicer if `style` and `script` inside `config` is in plural so it's consistent with bundle definition.

Fixed

> 
> One of the requested items was
> 
> > inject all or no scripts
> 
> how is this done current implementation?

It is defined in config.scripts value. That value is passed to 
```if (scripts.length > 0 && config.scripts) {...```
If the condition is true, all scripts will be loaded

Lanchi commented on src/utility/bundleInjector.js:
```diff
@@ -0,0 +1,78 @@
+    let styles = [];
+
+    if (bundle.scripts) {
+      scripts = filter(bundle.scripts, script => script.type === config.scriptType);
```
> It is defined in config.scripts value. That value is passed to
if (scripts.length > 0 && config.scripts) {...
If the condition is true, all scripts will be loaded

`bundle.scripts` get filtered by type considering `config.scriptType`.  So, if user wants to load only one type of the scripts, he will pass it as either `meta` or `component`. 
My question is: what should be passed if user wants to load both `meta` and `component` type scripts?

SoldoV replied:
From what I have understood, the original issue says you can import either component or meta scripts, not both:
> Builder imports only meta scripts, while other two projects import only components scripts.

So I have not implemented that. Have I misunderstood it?

Lanchi replied:
Feature request stated following:
> Bundle injector utility that will be created here should be configurable (e.g. config object parameter or something like that):
> 
> - inject all or no style sheets
> - inject all or no scripts
> - inject only meta or only components scripts
> - disregard deploy script type mentioned in vue-cli-plugin-chameleon, that's legacy

So, options regarding scripts are: only meta, only component, all (both types) or no scripts.

SoldoV replied:
Ok, I have deleted the ```scripts``` property and changed ```scriptStyle``` to ```scriptStyles``` array.
It has 3 possibilities:
1. [] - empty array, import no scripts
2. ['meta'] or ['components'] - import only one type of script
3. ['meta', 'components'] - import both script types

Lanchi replied:
You mean `scriptTypes`? 😄

SoldoV replied:
Yes, lapsus calami 😄

Lanchi replied:
Ok, just two more things and we're good to go: make `bundle` the first parameter, that is the object of `loadBundle` method after all; rename `config.scriptTypes` to `config.scripts` (consistent with bundle manifest).

SoldoV replied:
Changed
